tasks/12-ibm-codeengine-deploy.yaml

```diff
@@ -0,0 +1,150 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: ibm-codeengine-deploy
+spec:
+  params:
```

Comment:
All the values except for the image url are provided via the `codeengine-access` secret and that limits the usability of the task. (I.e. only one app can be deployed per CI namespace.) At the very least, the app name should be a required parameter like the image. Add optional parameters for the other values to allow any of the defaults to be overridden for a particular application. This can be done by defining the parameter with a default value of "", testing the parameter value in the task, and using the default value if one is not provided.